### PR TITLE
feat(presets): add `automergeStableNonMajor` preset

### PR DESCRIPTION
**Pull request by Turbo87**
## Changes

This PR adds a new `automergeStableMinor` preset.

## Context

I would like renovatebot to automerge semver-compatible changes automatically. I could use `automergeMinor` for this, but semver says that any version below `1.0.0` is allowed to include breaking changes in minor releases, or at least that is the way the `^` constraint in npm works.

This new preset can be combined with `automergePatch` and then it will automatically merge all patch updates and any minor updates that are semver-compatible.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

It includes a `description` field, so I assume no further documentation is necessary?

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

I've been running this as https://github.com/Turbo87/renovate-config/blob/116fdd4791f2a2e6a9f4692c36b779c62947ca62/automergeStableMinor.json for a while now and it's been working well.

I also have a [`automergeCaretConstraint`](https://github.com/Turbo87/renovate-config/blob/116fdd4791f2a2e6a9f4692c36b779c62947ca62/automergeCaretConstraint.json) preset in that repository, which combines the two presets mentioned above. Would it make sense to open a follow-up PR for that too?

### CODE REVIEW

File: lib/config/presets/internal/default.ts [hunk 1]
```diff
@@ -85,6 +85,17 @@ export const presets: Record<string, Preset> = {
...
     ignoreTests: false,
   },
+  automergeStableMinor: {
+    description:
+      'Automerge `minor` upgrades for semver stable packages if they pass tests.',
+    packageRules: [
+      {
+        automerge: true,
+        matchCurrentVersion: '>= 1.0.0',
+        matchUpdateTypes: ['minor'],
```
Comment (rarkins) [inline lib/config/presets/internal/default.ts:95]:
```suggestion
        matchUpdateTypes: ['minor', 'patch'],
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/presets/internal/default.ts (hunk 1 above)
Comment (Turbo87) [reply lib/config/presets/internal/default.ts:95]: hmm, should it be called `automergeStableMinorAndPatch` then?

Comment (rarkins) [reply lib/config/presets/internal/default.ts:95]: automergeStableNonMajor

Comment (Turbo87) [reply lib/config/presets/internal/default.ts:95]: Am I assuming correctly that the code suggestion above comes from the fact that the current `automergeMinor` also includes automerging for patch updates?

An alternative would be to implement `automergeCaretConstraint` directly and skip the intermediate step. Any thoughts on that?

Comment (rarkins) [reply lib/config/presets/internal/default.ts:95]: I'm not sure I follow

Comment (Turbo87) [reply lib/config/presets/internal/default.ts:95]: I was trying to understand why you would prefer to include `patch` in the `matchUpdateTypes` :)

Regarding `automergeCaretConstraint`, I was talking about https://github.com/Turbo87/renovate-config/blob/116fdd4791f2a2e6a9f4692c36b779c62947ca62/automergeCaretConstraint.json which I linked in the PR description above.

Comment (rarkins) [reply lib/config/presets/internal/default.ts:95]: I think people would expect both patch and minor types to be automerged. It's ok to have a short name if the description is accurate

File: lib/config/presets/internal/default.ts [hunk 2]
```diff
@@ -85,6 +85,17 @@ export const presets: Record<string, Preset> = {
     description: 'Require all status checks to pass before any automerging.',
     ignoreTests: false,
   },
+  automergeStableMinor: {
```
Comment (rarkins) [inline lib/config/presets/internal/default.ts:88]:
```suggestion
  automergeStableNonMajor: {
```

File: lib/config/presets/internal/default.ts [hunk 3]
```diff
@@ -85,6 +85,17 @@ export const presets: Record<string, Preset> = {
...
+    description:
+      'Automerge `minor` upgrades for semver stable packages if they pass tests.',
```
Comment (rarkins) [inline lib/config/presets/internal/default.ts:90]:
```suggestion
      'Automerge non-major upgrades for semver stable packages if they pass tests.',
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/presets/internal/default.ts [hunk 4]
```diff
@@ -85,6 +85,17 @@ export const presets: Record<string, Preset> = {
...
     ignoreTests: false,
   },
+  automergeStableNonMajor: {
+    description:
+      'Automerge non-major upgrades for semver stable packages if they pass tests.',
+    packageRules: [
+      {
+        automerge: true,
+        matchCurrentVersion: '>= 1.0.0',
+        matchUpdateTypes: ['minor'],
```
Comment (rarkins) [inline lib/config/presets/internal/default.ts:95]:
```suggestion
        matchUpdateTypes: ['minor', 'patch'],
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**